shopinvader_search_engine/models/shopinvader_backend.py

```diff
@@ -20,12 +20,31 @@ class ShopinvaderBackend(models.Model):
         help="Search Engine backend configuration to use",
     )
     index_ids = fields.One2many("se.index", related="se_backend_id.index_ids")
+    hide_se_settings = fields.Boolean(
```

Comment:
You are adding a new relation which seems useless to me. 
The flag should determine the behavior, not the other way around.

1. you select a se.backend
2. you state if it's shared by checking the flag